### DISCUSSION TITLE
VS Code: how to enable word based textual completions in typst files?

### DISCUSSION THREAD

**Discussion by MHellmund** [nvarner/typst-lsp#298] [Q&A]
I have
```
    "[latex]": {   "editor.wordBasedSuggestions": true },
``` 
in my VS code settings and it works great. Really speeds up my writing. But 
```
 "[typst]": { "editor.wordBasedSuggestions": true },
```
(or with `"[typst-lsp]"` ) does not work. Any ideas?

**Comment by nvarner** [comment]
I'm not sure why this doesn't work as expected, but `"[typst]"` is the right key to use, not `"[typst-lsp]"`.

**Comment by MHellmund** [comment]
Hm, I have only only one more observation.  When I search in vs code settings for `wordBasedSuggestions`, then some installed language extensions are listed:

![Screenshot at 2023-09-19 13-16-14](https://github.com/nvarner/typst-lsp/assets/1593619/b0da66a0-041f-4e86-b445-8348dbe20128)

Typst is not in the list. Perhaps the extension needs to register something? 

It is a really useful feature if you  do not write packages/templates but  documents.

**Reply by KagaJiankui** [reply]
It seems that the word based completions is enabled in **scripting** sections rather than **content** sections of a document.

**Comment by MHellmund** [comment]
I guess this is related: https://github.com/microsoft/vscode/issues/21611 

As suggested there, additional extensions like https://marketplace.visualstudio.com/items?itemName=Atishay-Jain.All-Autocomplete or  https://marketplace.visualstudio.com/items?itemName=RahulSinha.hippie-completion allow word based completion in the content part of a typst file.
